### PR TITLE
[14494] - Improve e2e testing docs and add cli args

### PR DESCRIPTION
**Pull request by v9ai**
## Description
Fixes to: https://github.com/WordPress/gutenberg/issues/14494
I added `local` arg to `npm run test-e2e` command.

## How has this been tested?
It was tested locally.

## Checklist:
- [x] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate.

### CODE REVIEW

File: packages/scripts/scripts/test-e2e.js [hunk 1]
```diff
@@ -50,4 +50,10 @@ if ( hasCliArg( '--puppeteer-interactive' ) ) {
 	process.env.PUPPETEER_SLOWMO = getCliArg( '--puppeteer-slowmo' ) || 80;
 }
 
+if ( hasCliArg( '--puppeteer-local' ) ) {
```
Comment (nicolad) [inline packages/scripts/scripts/test-e2e.js]: Hello @gziolo :)
I am not sure of 2 things here:
1. whether `local` is the best arg name
2. if checking arg with `--puppeteer-` as in example above is easy to reason about

File: packages/scripts/README.md [hunk 2]
```diff
@@ -215,6 +215,10 @@ _Example:_
 This is how you execute those scripts using the presented setup:
 
 * `npm run test:e2e` - runs all unit tests.
+* `npm run test-e2e interactive` - runs all unit tests interactively.
```
Comment (gziolo) [inline packages/scripts/README.md]: Shouldn't it look as follows? /cc @draganescu 
```md
* `npm run test-e2e -- --puppeteer-interactive` - runs all unit tests interactively.
* `npm run test-e2e FILE_NAME -- --puppeteer-interactive ` - runs one test file interactively.
* `npm run test-e2e:watch -- --puppeteer-interactive` - runs all tests interactively and watch for changes.
```

In watch mode, you rather don't pass the name of the file as by default it runs only for modified files.

File: packages/scripts/scripts/test-e2e.js (hunk 1 above)
Comment (gziolo) [reply packages/scripts/scripts/test-e2e.js]: Actually, the defaults are:

```js
process.env.WP_BASE_URL = 'http://localhost:8889';
process.env.WP_USERNAME = 'admin';
process.env.WP_PASSWORD = 'password';
```
See: https://github.com/WordPress/gutenberg/blob/master/packages/e2e-test-utils/src/shared/config.js#L9

So there is no need to set them explicitly. We need the opposite, to have a way to override all those 3 values:



```js
if ( hasCliArg( '--wordpress-host' ) ) {
    process.env.WP_BASE_URL = getCliArg( '--wordpress-host' );
}
if ( hasCliArg( '--wordpress-username' ) ) {
    process.env. WP_USERNAME = getCliArg( '--wordpress-username' );
}
if ( hasCliArg( '--wordpress-password' ) ) {
    process.env. WP_PASSWORD = getCliArg( '--wordpress-password' );
}
```
or even better something which uses look up table and loops over it.

**Review by gziolo — CHANGES_REQUESTED** [review]

File: packages/scripts/README.md (hunk 2 above)
Comment (nicolad) [reply packages/scripts/README.md]: I made this change based on this PR's description: https://github.com/WordPress/gutenberg/pull/14129

Comment (draganescu) [reply packages/scripts/README.md]: yes, but I didn't update the PR's description with the changes in the comments. @gziolo is right, in the end it was developed as:
```
* `npm run test-e2e -- --puppeteer-interactive` - runs all unit tests interactively.
* `npm run test-e2e FILE_NAME -- --puppeteer-interactive ` - runs one test file interactively.
* `npm run test-e2e:watch -- --puppeteer-interactive` - runs all tests interactively and watch for changes.
```

**Comment by nicolad** [comment]
I messed up deleting the fork, created another PR based on current feedback: https://github.com/WordPress/gutenberg/pull/14717

**Pull request closed without merging**